docs/monitoring.md

```diff
@@ -923,13 +923,17 @@ This is the component with the largest amount of instrumented metrics
   - memory.remainingOnHeapMem_MB
 
 - namespace=HiveExternalCatalog
+  - **note:**: these metrics are conditional to a configuration parameter:
+    `spark.metrics.register.static.sources.enabled` (default is true) 
   - fileCacheHits.count
   - filesDiscovered.count
   - hiveClientCalls.count
   - parallelListingJobCount.count
   - partitionsFetched.count
 
 - namespace=CodeGenerator
+  - **note:**: these metrics are conditional to a configuration parameter:
+    `spark.metrics.register.static.sources.enabled` (default is true) 
```

Comment:
`Executor` also has the following. I guess this PR aims to control them together.
```
- namespace=HiveExternalCatalog
  - fileCacheHits.count
  - filesDiscovered.count
  - hiveClientCalls.count
  - parallelListingJobCount.count
  - partitionsFetched.count
- namespace=CodeGenerator
  - compilationTime (histogram)
  - generatedClassSize (histogram)
  - generatedMethodSize (histogram)
  - hiveClientCalls.count
  - sourceCodeSize (histogram)
```